core/commands/swarm.go

```diff
@@ -61,6 +62,140 @@ const (
 	swarmDirectionOptionName = "direction"
 )
 
+var swarmPeeringCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "modify the peering service.",
+		ShortDescription: `
+'ipfs swarm peering' is a tool to manupulate the peering service. 
+Peers in the peering service is maintained to be connected, reconnected 
+on disconnect with a back-off.
+`,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"add": swarmPeeringAddCmd,
+		"ls":  swarmPeeringLsCmd,
+		"rm":  swarmPeeringRmCmd,
+	},
+}
+
+var swarmPeeringAddCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "add peers into the peering service.",
+		ShortDescription: `
+'ipfs swarm peering add' adds peers into the peering service.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("address", true, true, "address of peer to add into the PeeringService"),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		addrs := make([]ma.Multiaddr, len(req.Arguments))
+
+		for i, arg := range req.Arguments {
+			addr, err := ma.NewMultiaddr(arg)
+			if err != nil {
+				return err
+			}
+
+			addrs[i] = addr
+		}
+
+		addInfos, err := peer.AddrInfosFromP2pAddrs(addrs...)
+		if err != nil {
+			return err
+		}
+
+		node, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+
+		output := make([]string, len(addInfos))
+		for i, id := range addInfos {
+			output[i] = "add " + id.ID.Pretty()
+			node.Peering.AddPeer(id)
+
+			output[i] += " success"
+		}
+
+		return cmds.EmitOnce(res, &stringList{output})
+	},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(stringListEncoder),
+	},
+	Type: stringList{},
+}
+
+var swarmPeeringLsCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "list peers registered in the peering service.",
+		ShortDescription: `
+'ipfs swarm peering ls' lists peers registered in the peering service.`,
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		node, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		peers := node.Peering.ListPeer()
+		return cmds.EmitOnce(res, addrInfos{Peers: peers})
```

Comment:
Let's emit each result as a separate AddrInfo (no need to batch them).

Reply:
That is, call `res.Emit` on each one.

(I know our other commands tend to batch them, but that's more of an historical mistake than anything)

Reply:
I had overlooked this comment. I'll fix it now.

Reply:
Because `ListPeer()` returns a slice, it does not sound bad to me to return `peers` with `EmitOnce`. How do you think?

Reply:
The main concern is if the list grows long and we need to stream it. But honestly, it shouldn't be an issue.

LGTM.

Reply:
I've understood the problem. Thank you for the comment.